lib/sqlalchemy/sql/schema.py

```diff
@@ -4549,7 +4575,10 @@ def _set_parent(self, parent: SchemaEventTarget, **kw: Any) -> None:
 
         for c in self._columns:
             c.primary_key = True
-            if c._user_defined_nullable is NULL_UNSPECIFIED:
+            if (
+                not isinstance(c, Period)
+                and c._user_defined_nullable is NULL_UNSPECIFIED
+            ):
                 c.nullable = False
         if table_pks:
```

Comment:
**Trevor Gross** (tgross35) wrote:

Ack

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/3803